libs/wingsdk/src/helpers.ts

```diff
@@ -1,9 +1,13 @@
 // Code in this file will be automatically included in all inflight code bundles,
 // so avoid importing anything heavy here.
-import { deepStrictEqual, notDeepStrictEqual } from "node:assert";
 import * as path from "node:path";
 import type { Construct } from "constructs";
 import type { Node } from "./std/node";
+// since we moved from node:18 to node:20 the deepStrictEqual doesn't work as expected.
+// https://github.com/winglang/wing/issues/4444
+// therefore we're using an npm package called "assert" instead.
+// eslint-disable-next-line @typescript-eslint/no-require-imports
+const { deepStrictEqual, notDeepStrictEqual } = require("external-assert");
```

Comment:
I don't think it's worthwhile to take on this new dependency just for deepStringEqual. This `helpers.ts` file is included in every inflight/preflight so this module will end up being bundled everywhere.

Instead it may be simple enough to implement/vendor the function ourselves.
Here's an example in [vitest](https://github.com/vitest-dev/vitest/blob/d6700bbd895e63776263b206ec73ccbb858a1b94/packages/expect/src/jest-utils.ts#L76) (some of it can be easily removed)

Worst case scenario it may be best to find a package that only deals with `deepStringEqual` instead of the entire `assert` module.

Reply:
took the function stright from node 18: https://github.com/nodejs/node/blob/v18.x/lib/internal/util/comparisons.js
(and added dozens of deepEquality tests, thanks for chatgpt)